Brokerages/InteractiveBrokers/InteractiveBrokersBrokerage.cs

```diff
@@ -1404,18 +1396,10 @@ private void HandleExecutionDetails(object sender, IB.ExecutionDetailsEventArgs
                     return;
                 }
 
-                var currentQuantityFilled = Convert.ToInt32(executionDetails.Execution.Shares);
-                var totalQuantityFilled = Convert.ToInt32(executionDetails.Execution.CumQty);
-                var remainingQuantity = Convert.ToInt32(order.AbsoluteQuantity - totalQuantityFilled);
-                var price = Convert.ToDecimal(executionDetails.Execution.Price);
+                // save execution in dictionary and wait for commission report,
+                // so we can fire the order event for the fill (or partial fill) with the actual IB fees
 
-                // set order status based on remaining quantity
-                var status = remainingQuantity > 0 ? OrderStatus.PartiallyFilled : OrderStatus.Filled;
-
-                if (!IsFillAlreadyHandled(order, executionDetails.Execution.OrderId, remainingQuantity))
-                {
-                    HandleFill(order, currentQuantityFilled, totalQuantityFilled, remainingQuantity, price, status);
-                }
+                _orderExecutions[executionDetails.Execution.ExecId] = executionDetails.Execution;
```

Comment:
Are we guaranteed to receive the fill *before* the commission event? Any documentation that provides such guarantee? I'm thinking of receiving the commission event first and then this one and we'll never emit the fill (unless I've ready this incorrectly)

Reply:
This is the only documentation page regarding these two events (it doesn't mention specific order of events):
https://interactivebrokers.github.io/tws-api/executions_commissions.html#gsc.tab=0
While testing I have never seen a commission report received before the execution it refers to (by ExecId).

I agree it is safer not to assume the current ordering, I'll change the logic to support receiving the events in any order.